### PR TITLE
[Kotlin migration] Statistics

### PR DESCRIPTION
**Pull request by lukstbit**
## Purpose / Description
This PR migrates the Statistics.java file to kotlin following the kotlin migration guide. The migration left more things to be improved later, I've added several `@KotlinCleanup` annotations in the code.

## How Has This Been Tested?

Ran tests, manually checked the Statistics screen in the application.

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/Statistics.kt [hunk 1]
```diff
@@ -14,646 +14,542 @@
...
+        super.onCollectionLoaded(col)
 
         // Create the adapter that will return a fragment for each of the three
         // primary sections of the activity.
         // Set up the ViewPager with the sections adapter.
-        mViewPager = findViewById(R.id.pager);
-        mViewPager.setAdapter(new Statistics.StatsPagerAdapter((this)));
-        mViewPager.setOffscreenPageLimit(8);
-        mSlidingTabLayout = findViewById(R.id.sliding_tabs);
+        viewPager = findViewById(R.id.pager)
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/Statistics.kt:85]: cleanup: scope function

File: AnkiDroid/src/main/java/com/ichi2/anki/Statistics.kt [hunk 2]
```diff
@@ -14,646 +14,542 @@
...
     }
 
-
-    @Override
-    public boolean onCreateOptionsMenu(Menu menu) {
-        super.onCreateOptionsMenu(menu);
-        //System.err.println("in onCreateOptionsMenu");
-        MenuInflater inflater = getMenuInflater();
-        inflater.inflate(R.menu.anki_stats, menu);
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/Statistics.kt:130]: cleanup: `findItem -> isChecked` to nested function

File: AnkiDroid/src/main/java/com/ichi2/anki/Statistics.kt [hunk 3]
```diff
@@ -14,646 +14,542 @@
...
-                case CARDS_TYPES_TAB_POSITION:
-                    mStatisticsTask = taskHandler.createChart(
-                            Stats.ChartType.CARDS_TYPES, mChart, mProgressBar);
-                    break;
+        @KotlinCleanup("improve when statement code")
+        private fun createChart() {
+            val statisticsActivity = requireActivity() as Statistics
+            val taskHandler = statisticsActivity.taskHandler
+            when (mSectionNumber) {
+                FORECAST_TAB_POSITION -> statisticsTask = taskHandler!!.createChart(
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/Statistics.kt:400]: nested function: `taskHandler!!.createChart`, or have the `ChartType` returned from the when

**Review by david-allison — COMMENTED** [review]
This is great! 

A few more cleanups to add, but the conversion looks great, thanks so much

**Comment by lukstbit** [comment]
@david-allison I've added the extra `@KotlinCleanup` requested.

**Review by ShridharGoel — APPROVED** [review]

**Pull request merged**